### DISCUSSION TITLE
Is XLA parallel compilation necessary?

### DISCUSSION THREAD

**Discussion by Kajiih** [jax-ml/jax#17754] [Q&A]
Hello dead JAX community!

I am trying JAX on a remote server for which I can't update cuda version, and I got this warning when using JAX on GPU:
```bash
"The NVIDIA driver's CUDA version is 11.2 which is older than the ptxas CUDA version (11.8.89). Because the driver is older than the ptxas version, XLA is disabling parallel compilation, which may slow down compilation. You should update your NVIDIA driver or use the NVIDIA-provided CUDA forward compatibility packages."
```
I intend to use JIT compilation, but I don't have a lot of experience with XLA so I'm not sure of the impact it will have.

Can I keep my current installation as is without a too significant loss of performance? Like JAX will still run on GPU, the only difference with updated CUDA version will be the compilation time, right? (compilation time on CPU is not too big here)

Also, I have seen this some answers in this issue #5231 related to this, but is there still an incompatibility between JAX and Pytorch on GPU when installed in the same virtual environment? (Can't use GPUs when torch is also installed)

Thank you!

**Comment by ParamB11** [comment]
Found this answer in another issue:
https://github.com/google/jax/issues/18027#issuecomment-1758108802
